src/dialects/abstract/query-interface.d.ts

```diff
@@ -214,6 +214,15 @@ export interface IndexesOptions {
 
 export interface QueryInterfaceIndexOptions extends IndexesOptions, QueryInterfaceOptions {}
 
+/**
+ * Interface for bulk delete options
+ */
+export interface QueryInterfaceBulkDeleteOptions {
+  truncate?: boolean
+  cascade?: boolean
+  restartIdentity?: boolean
```

Comment:
in non-truncate mode, `limit` and `offset` (db2) are accepted

Reply:
Is there some TypeScript magic that would be useful for only allowing `limit` and `offset` when `truncate` is false?

Reply:
Nothing very clean unfortunately. I would just add them and document in which mode they apply

We can fix this in the future by removing truncate ability from `.bulkDelete`, and have `.truncate` instead. It's two very different SQL operations after all